### ISSUE TITLE
save tensorflow  model to .pb file 

### ISSUE THREAD

**Issue by omrylcn** [tensorflow/tensorflow#22579]
Hi I wonder , how do I save model to .pb file to retrain model  ? Below ,there is sample model for experiment. Thanks for helping

#######################################################################
x=tf.placeholder(tf.float32,shape=[None,2],name="x")
y=tf.placeholder(tf.float32,shape=[None,1],name="y")

with tf.name_scope("dnn"):
      layer1=tf.layers.dense(x,15,activation=tf.nn.relu,
                           kernel_initializer= tf.initializers.truncated_normal(),name="layer1")
    layer2=tf.layers.dense(layer1,20,activation=tf.nn.relu,
                           kernel_initializer= tf.initializers.truncated_normal(),name="layer2")
    logits=tf.layers.dense(layer2,1,
                           kernel_initializer= tf.initializers.truncated_normal(),name="logits")
    out=tf.nn.sigmoid(logits,name="out")
    
with tf.name_scope("train"):
    loss=tf.losses.sigmoid_cross_entropy(multi_class_labels=y,logits=logits)
    train_op=tf.train.AdamOptimizer().minimize(loss)

with tf.name_scope("eval"):
    correct=tf.equal(tf.round(out),y)
    accuracy = tf.reduce_mean(tf.cast(correct, tf.float32))

saver=tf.train.Saver()

epochs=2000
with tf.Session( ) as sess:
    sess.run(tf.global_variables_initializer())
    for epoch in range(epochs+1):
        sess.run(train_op,feed_dict={x:trainx,y:trainy})
        if epoch%1000==0:
            l,a=sess.run([loss,accuracy],feed_dict={x:trainx,y:trainy})
            print("Epoch {} | Accuracy : {:.2f} | Loss :{:.2f}".format(epoch,a,l))
            saver.save(sess,"./moon_model")
    tf.train.write_graph(sess.graph.as_graph_def(), '.', 'moon_model.pbtxt', as_text=True)

#########################################################################3

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device
